saleor/attribute/migrations/tasks/saleor3_16.py

```diff
@@ -0,0 +1,49 @@
+from ....celeryconf import app
+from ...models import AssignedProductAttributeValue
+
+from django.db import connection
+
+# batch size to make sure that task is completed in 1 second
+# and we don't use too much memory
+BATCH_SIZE = 5000
+
+
+def update_product_assignment():
+    """Assign product_id to a new field on assignedproductattributevalue.
+
+    Take the values from attribute_assignedproductattribute to product_id.
+    The old field has already been deleted in Django State operations so we need
+    to use raw SQL to get the value and copy the assignment from the old table.
+    """
+    with connection.cursor() as cursor:
+        cursor.execute(
```

Comment:
Are you sure that this is the correct approach? Previously we were doing such actions in patch releases.
So assuming that the newest release is 3.15.0, and main is unreleased 3.16. 
- 3.15.1 should receive migrations, which will move the data to the new fields
- 3.16.0 could receive alterField to remove `null=True`.

---
With the current approach, we have a situation, in which we already removed the field from Django State but we haven't triggered the task yet. It means that 3.16 workers (which operate on new code) will not have details mandatory to correctly display the data. 

---

Some stores can have thousands of attributes, which means that until fully processing the celery task we could display corrupted data, am I correct?